pkg/backend/filestate/state.go

```diff
@@ -302,16 +321,38 @@ func (b *localBackend) backupStack(name tokens.Name) error {
 	stackFile := filepath.Base(stackPath)
 	ext := filepath.Ext(stackFile)
 	base := strings.TrimSuffix(stackFile, ext)
+	if ext2 := filepath.Ext(base); ext2 != "" && ext == encoding.GZIPExt {
+		// base: stack-name.json, ext: .gz
+		// ->
+		// base: stack-name, ext: .json.gz
+		ext = ext2 + ext
+		base = strings.TrimSuffix(base, ext2)
```

Comment:
```suggestion
		base = strings.TrimSuffix(base, ext2)
		byts = encoding.Gzip(byts)
```
Pulumi works great without this fix as it checks the magic header instead of the file extension to know if a file is gzip'ed. But currently `~/.pulumi/backups/dummy/dummy.1645191346109283284.json.gz` would be written as plaintext json.

Reply:
@awoimbee I don't understand. I'm going to explain my though process. Can you tell me where I went wrong?

 We read from the bucket a file named `stackPath`. We then write it to `backupDir/stackPath`. If the file is `myStack.json.gz`, we then write it directly to `stackPath/myStack.json.gz`. We shouldn't need to compress it since it is already a zip file (since it is named `myStack.json.gz`).

Reply:
Oh yeah, I think you're right. I have some weird files with `.gz` that aren't gzip, but that might be caused by a debug build I ran (and anyways pulumi can read it fine).